app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id_id
+  has_many :comments, class_name: 'Comment', foreign_key: :post_id, dependent: :destroy
+  has_many :likes, class_name: 'Like', foreign_key: :post_id, dependent: :destroy
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] Great work on defining the methods for your post model! 😊 To enhance the maintainability and organization of your code, I recommend considering the use of Rails Active Callbacks, specifically `after_save`, to automatically trigger the `update_posts_counter` method each time a new post is saved to the database. This approach will help you keep your code clean and ensure that the counter is always up to date without manual intervention.

### Below is an example of making use of `after_save`:
```
after_save :update_posts_counter

def update_posts_counter
    author.increment!(:posts_counter)
end
```

Keep up the good work! 👍🚀

Reply:
Hi @Strangeal! Oh, nice 🎯, thank you for the suggestion! I will totally implement it now. 🥂